src/app/api/file/route.ts

```diff
@@ -0,0 +1,130 @@
+import { NextResponse } from "next/server";
+import { File, FileResponse } from "./route.schema";
+import { prisma } from "@server/db/client";
+import { withSession } from "@server/decorator";
+import { createDriveService } from "@server/service";
+
+export const POST = withSession(async (request) => {
+  try {
+    const service = await createDriveService(request.session.user.id);
+
+    const body = await request.req.json();
+    const fileRequest = File.safeParse(body);
+
+    if (!fileRequest.success) {
+      console.log(fileRequest.error);
+      return NextResponse.json(
+        FileResponse.parse({
+          code: "NOT_FOUND",
+          message: "Unsuccessful request creation",
+        }),
+        { status: 400 }
+      );
+    } else {
+      const fileData = fileRequest.data;
+
+      // Check that user has this senior assigned to them
+      const user = await prisma.user.findFirst({
+        where: {
+          id: request.session.user.id,
+        },
+      });
+
+      if (user === null || user.SeniorIDs === null) {
+        return NextResponse.json(
+          FileResponse.parse({
+            code: "NO_USER",
+            message: "User does not exist",
+          })
+        );
+      }
+
+      if (
+        !user.SeniorIDs.some(
+          (seniorId: string) => seniorId === fileData.seniorId
+        )
+      ) {
+        return NextResponse.json(
+          FileResponse.parse({
+            code: "NOT_AUTHORIZED",
+            message: "Senior not assigned to user",
+          }),
+          { status: 404 }
+        );
+      }
+
+      // Get senior from database
+      const foundSenior = await prisma.senior.findUnique({
+        where: { id: fileData.seniorId },
+      });
+      if (foundSenior == null) {
+        return NextResponse.json(
+          FileResponse.parse({
+            code: "NOT_FOUND",
+            message: "Senior not found",
+          }),
+          { status: 404 }
+        );
+      }
+
+      const parentID = foundSenior.folder;
+
+      const formatted_date =
+        (fileData.date.getMonth() > 8
+          ? fileData.date.getMonth() + 1
+          : "0" + (fileData.date.getMonth() + 1)) +
+        "/" +
+        (fileData.date.getDate() > 9
+          ? fileData.date.getDate()
+          : "0" + fileData.date.getDate()) +
+        "/" +
+        fileData.date.getFullYear();
+
+      const fileMetadata = {
+        name: [formatted_date],
+        mimeType: "application/vnd.google-apps.document",
+        parents: [parentID],
+      };
+
+      const fileCreateData = {
+        resource: fileMetadata,
+        fields: "id",
+      };
+
+      // NOTE: File will still be created on Drive even if it fails on MongoDB
```

Comment:
Good note. In a more robust system, we should make google doc and mongodb action a transaction. We can talk more about this in the next meeting